src/amber/cli/templates/app/package.json.ecr

```diff
@@ -12,16 +12,20 @@
     "test": "echo \"No test specified\" && exit 1"
   },
   "devDependencies": {
-    "babel-core": "^6.26.0",
-    "babel-loader": "^7.1.2",
-    "babel-preset-env": "^1.6.1",
+    "@babel/core": "^7.0.0-beta.38",
```

Comment:
Why we are using beta here?

> My hope was to keep the web pack and npm packages to a bare minimum. This adds two more npm packages and a plugin that is a beta version. Is there a way to do this without the extra packages?

@drujensen ^ https://github.com/amberframework/amber/pull/301#issuecomment-335983603

Reply:
BTW, this doen't add 2 packages but 9 package dependencies 😅

Reply:
@faustinoaq I tried to closely match the npm dependencies that come default in rails/webpacker.  However, the amber.js webpack config was breaking with the latest version of babel-core/babel-loader/babel-preset-env (at least on my machine), and I couldn't find anything to fix it other than upgrading as shown in the babel-loader docs on github.  Feel free to modify the babel dependencies to a different configuration.  I will not be offended :).

Reply:
Also, my experience with customizing webpack has been terrible.  Coming to Amber, I would have much preferred a nice setup that does for me everything that Sprockets does in rails. That's just how I feel, so I decided to configure it that way.  Also, these dependencies aren't really dependencies because they are only in the generated template, not in the actual compiled code.  Thus, anyone can customize their webpack however much they want.  It just means that new projects come with a nice default.  Those are my thoughts, but I'm also new to Amber and I know my thoughts are just opinion :).

Reply:
nice catch @faustinoaq.  Let's see if we can get this to work without the `beta`.

Reply:
@faustinoaq I'm looking at this commit.  https://github.com/amberframework/amber/commit/76420ea0f6f2575a96c7722e8a14a476e625b728#diff-fbea2c0ca78b494fd0f7e012198847d4.  I can't get npm run build to succeed with this set of babel dependencies when including "import Amber from 'amber'" in the main.js.  The error is: ERROR in ../amber/assets/js/amber.js
Module build failed: Error: Couldn't find preset "env" relative to directory "/Users/joshjuncker/Workspace/amber/assets/js".  This doesn't happen with the newer beta versions.  And these older versions are being archived.  Look up babel-preset-env and you'll see that it is archived.  Can you help me figure out how to make it work?  Or if we just decide to use the beta.  Either way, I just can't figure it out :)

Reply:
Oh, you're right, official webpack docs are using beta versions https://webpack.js.org/loaders/babel-loader/#install Maybe we don't need to worry  about this 😅

Reply:
seriously??!!  The JS world is such a mess.  Let's make sure Crystal and Amber don't head down this path.

Reply:
Sorry, we can use webpack 3.x babel-loader 7.x | babel 6.x Can we use this:

```
npm install babel-loader babel-core babel-preset-env
```

Reply:
@faustinoaq I've followed that exactly and get the error referenced in my comment above.  Are you able to get it to work?

Reply:
> Are you able to get it to work?

Yes, I do, I'm using non-beta version and I can't reproduce your issue:

```
➜  app npm install  
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@1.1.3 (node_modules/fsevents):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@1.1.3: wanted {"os":"darwin","arch":"any"} (current: {"os":"linux","arch":"x64"})

up to date in 7.772s
➜  app npm ls --depth=0
app@0.1.0 /home/main/Projects/app
├── babel-core@6.26.0
├── babel-loader@7.1.2
├── babel-preset-env@1.6.1
├── clean-webpack-plugin@0.1.17
├── compression-webpack-plugin@1.1.3
├── css-loader@0.28.9
├── extract-text-webpack-plugin@3.0.2
├── file-loader@1.1.6
├── node-sass@4.7.2
├── require-context@1.0.2
├── sass-loader@6.0.6
├── style-loader@0.19.1
├── webpack@3.10.0
├── webpack-manifest-plugin@1.3.2
└── webpack-merge@4.1.1
➜  app npm run watch

> app@0.1.0 watch /home/main/Projects/app
> webpack -w --config config/webpack/development.js

clean-webpack-plugin: /home/main/Projects/app/public/dist has been removed.

Webpack is watching the files…
```

Also works well if I use beta dependencies that you're using:

```
➜  app rm -rf node_modules 
➜  app npm install        

> node-sass@4.7.2 install /home/main/Projects/app/node_modules/node-sass
> node scripts/install.js

Cached binary found at /home/main/.npm/node-sass/4.7.2/linux-x64-59_binding.node

> uglifyjs-webpack-plugin@0.4.6 postinstall /home/main/Projects/app/node_modules/uglifyjs-webpack-plugin
> node lib/post_install.js


> node-sass@4.7.2 postinstall /home/main/Projects/app/node_modules/node-sass
> node scripts/build.js

Binary found at /home/main/Projects/app/node_modules/node-sass/vendor/linux-x64-59/binding.node
Testing binary
Binary is fine
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@1.1.3 (node_modules/fsevents):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@1.1.3: wanted {"os":"darwin","arch":"any"} (current: {"os":"linux","arch":"x64"})

added 681 packages in 39.883s
➜  app npm ls --depth=0   
app@0.1.0 /home/main/Projects/app
├── @babel/core@7.0.0-beta.38
├── @babel/preset-env@7.0.0-beta.38
├── babel-loader@8.0.0-beta.0
├── clean-webpack-plugin@0.1.17
├── compression-webpack-plugin@1.1.3
├── css-loader@0.28.9
├── extract-text-webpack-plugin@3.0.2
├── file-loader@1.1.6
├── node-sass@4.7.2
├── require-context@1.0.2
├── sass-loader@6.0.6
├── style-loader@0.19.1
├── webpack@3.10.0
├── webpack-manifest-plugin@1.3.2
└── webpack-merge@4.1.1
➜  app npm run watch      

> app@0.1.0 watch /home/main/Projects/app
> webpack -w --config config/webpack/development.js

clean-webpack-plugin: /home/main/Projects/app/public/dist has been removed.

Webpack is watching the files…
```

I'm using the same `common.js` as described in this PR

I don't know why your webpack build is failing, the presets `env` (non-beta) and `@babel/preset-env` (beta) works just fine for me 😅

Reply:
@faustinoaq Are you including Amber in your main.js?  It isn't the common js reference that is failing.  It is because I include amber in my main js and amber.js has its own webpack config with it.  It works fine if I don't include amber.

Reply:
> Are you including Amber in your main.js?

Yes, this is my `main.js` : (Other files like `run_macros/` dir and Assets Helper also included, I'm using this PR)

```js
var context = require.context('../images', true) // Necessary for asset helper.  Has webpack import all the images regardless of use in js/css.
var context = require.context('../fonts', true) // Necessary for asset helper.  Has webpack import all the fonts regardless of use in js/css.
import Amber from 'amber'

console.log("it works!");
```

But I'm having another problem, caching system is not working well, changes aren't reflected in the browser, it always shows the first compiled assets, it watches and compiles files but browser keep unchanged (fetching cached file):

![vokoscreen-2018-01-25_09-35-11](https://user-images.githubusercontent.com/3067335/35393729-d657e3cc-01b3-11e8-960c-1002d1dcce36.gif)

Maybe we need to tweak `Amber::Pipe::Reload` too 😅